srcts/src/components/darkModeSwitch.ts

```diff
@@ -0,0 +1,323 @@
+import type { CSSResultGroup } from "lit";
+import { LitElement } from "lit";
+import { css, html } from "lit";
+import { property } from "lit/decorators.js";
+
+// Inspired by:
+// https://web.dev/building-a-theme-switch-component/
+// https://web.dev/patterns/theming/theme-switch/
+// https://github.com/argyleink/gui-challenges/tree/main/theme-switch
+
+export type DarkModeMessageToggle = {
+  method: "toggle";
+  value: DarkModeSwitch["mode"] | undefined;
+};
+
+export class DarkModeSwitch extends LitElement {
+  static isShinyInput = true;
+  static tagName = "bslib-input-dark-mode";
+  static shinyCustomMessageHandlers = {
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    "bslib.toggle-dark-mode": ({
+      method,
+      value,
+    }: DarkModeMessageToggle): void => {
+      // Similar to DarkModeSwitch.receiveMessage(), but we directly update the
+      // Bootstrap attribute on the <html> element. Currently, all toggle switches
+      // follow this value.
+
+      if (method !== "toggle") return;
+
+      if (typeof value === "undefined" || value === null) {
+        const current = document.documentElement.dataset.bsTheme || "light";
+        value = current === "light" ? "dark" : "light";
+      }
+
+      document.documentElement.dataset.bsTheme = value;
+    },
+  };
+
+  private attribute = "data-shinytheme";
```

Comment:
The default value is set both here and inside of `connectedCallback()`. There are two ways I can think of to simplify, both of which are defensible:

One is to use `@property` here:

```suggestion
  @property({ type: String})
  private attribute = "data-shinytheme";
```

And then you don't have to do manually set it in `connectedCallback`.

The other option is to modify `connectedCallback` so instead of this:

```ts
    this.attribute = this.getAttribute("attribute") || "data-shinytheme";
```

it has this:

```ts
    this.attribute = this.getAttribute("attribute") || this.attribute;
```

Reply:
We've add bslib-specific naming in many more places than I originally intended that at this point I think this could be a static value. Regardless, I opted for the second suggestion – a private value with an initial value that's possibly overwritten when the component is connected.